### ISSUE TITLE
AVIF version 10.0.1 broken in avifImageYUVToRGB requiring more than two arguments

### ISSUE THREAD

**Issue by TopRockPhoto** [darktable-org/darktable#12425]
**Describe the bug/issue**

METHOD A
**To Reproduce**
upgrade AVIF to version 0.10.1, as done on your distribution.

Then attempt an export to avif  at any bit depth, (I tried 12-bit and 8-bit), and at any compression, (I tried lossless and lossy @ 81%)
Darktable shuts down.

**Expected behavior**
After a few seconds (about 20 in my case with my system), the file is exported.

METHOD B
**To Reproduce**
upgrade AVIF to version 0.10.1, as done on your distribution.
Get the source code to darktable 4.0.0 (or 4.0.1),
Do cmake 
Do make

Error generated saying too few arguments on avifImageYUVtoRGB

`[ 52%] Building CXX object bin/CMakeFiles/lib_darktable.dir/common/imageio_exr.cc.o
[ 52%] Building C object bin/CMakeFiles/lib_darktable.dir/common/imageio_avif.c.o
[ 53%] Building C object bin/CMakeFiles/lib_darktable.dir/common/imageio_heif.c.o
[ 53%] Building C object bin/CMakeFiles/lib_darktable.dir/common/imageio_libraw.c.o
[ 53%] Building C object bin/CMakeFiles/lib_darktable.dir/common/imageio_j2k.c.o
[ 53%] Building C object bin/CMakeFiles/lib_darktable.dir/common/pwstorage/backend_libsecret.c.o
[ 53%] Building C object bin/CMakeFiles/lib_darktable.dir/common/sqliteicu.c.o
[ 53%] Building C object bin/CMakeFiles/lib_darktable.dir/common/camera_control.c.o
[ 53%] Building C object bin/CMakeFiles/lib_darktable.dir/control/jobs/camera_jobs.c.o
[ 53%] Building C object bin/CMakeFiles/lib_darktable.dir/lua/cairo.c.o
[ 53%] Building C object bin/CMakeFiles/lib_darktable.dir/lua/call.c.o
[ 53%] Building C object bin/CMakeFiles/lib_darktable.dir/lua/configuration.c.o
/home/karim/Downloads/AV/Photography/darktable/darktable/src/common/imageio_avif.c: In function ‘dt_imageio_open_avif’:
/home/karim/Downloads/AV/Photography/darktable/darktable/src/common/imageio_avif.c:74:12: error: too few arguments to function ‘avifImageYUVToRGB’
   74 |   result = avifImageYUVToRGB(avif, &rgb);
      |            ^~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
make[2]: *** [bin/CMakeFiles/lib_darktable.dir/build.make:1947: bin/CMakeFiles/lib_darktable.dir/common/imageio_avif.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:2593: bin/CMakeFiles/lib_darktable.dir/all] Error 2
make: *** [Makefile:171: all] Error 2
`

**Expected behavior**
build complete.

*Error introduced*
With AVIF version 0.9.3, there is no error.
With AVIF version 0.10.1 the error is present.
AVIF 10.0.0 was not tested.

This is NOT an error per se with the darktable code, but an error due to changes within AVIF and the avifImageYUVToRGB library which has been updated to use more arguments.

*Workaround*
Downgrade AVIF to version 0.9.3 (0.10.0 was not tested).
Old compiled binaries will start to work. New builds will compile.

**Platform**

* darktable version : e.g. 4.0.0 & 4.0.1
* OS                : e.g. Linux - kernel 5..15.0-47-generic (x86_64) / Ubuntu 22.04.1
* Linux - Distro    : e.g. Ubuntu 22.04.1
* Memory            :  32GiB
* Graphics card     :  nVidia GeForce GTX 760/PCIe/SSE2
* Graphics driver   :  4.6.0 nVidia 470.141.03
* OpenCL installed  : OpenCL 3.0 CUDA 11.4.264
* OpenCL activated  : yes
* Xorg              : x11 11.0 version 1.21.1.3 (12101003)
* Desktop           : GNOME 42.4.0  (Ubuntu)
* GTK+              : 2.99.3
* gcc               : 11.2.0 (x86_64-linux-gnu)
* cflags            : defaults
* CMAKE_BUILD_TYPE  : release

**Comment by ptilopteri** [comment]
running openSUSE Tumbleweed with
libavif14-0.10.1-2.2.x86_64
darktable 4.1.0~git272.5a1a1845-8238.1

just exported an image to save as avif at 8,10,12 bit
and all completed successfully w/o any crash.

images available for your purvue
  http://wahoo.no-ip.org/~paka/200119_112402_1image_01.avif
  http://wahoo.no-ip.org/~paka/200119_112402_1image_02.avif
  http://wahoo.no-ip.org/~paka/200119_112402_1image.avif

perhaps you have a local problem.

**Comment by ptilopteri** [comment]
seeing that many have viewed the avif files generated as noted above,
can we probably close this issue now?

**Comment by kmilos** [comment]
There is no change in this function from 0.9.3 to 0.10.1: https://github.com/AOMediaCodec/libavif/compare/v0.9.3...v0.10.1#diff-0568c669c075285d7f120551ab6e63b11bf7b138ed1cbd6e2fcc5c79793012fe

Packaging issue on your distro?

**Comment by KarlMagnusLarsson** [comment]
I get the same error when building Darktable master from source on Debian 11.4, using instruction from [here](https://github.com/darktable-org/darktable/wiki/Getting-started).

` scbi --prefix=/home/magnus/test/darktable  --deps --update c-darktable`

Build fails with same error as stated by @TopRockPhoto:

`[ 48%] Building C object bin/CMakeFiles/lib_darktable.dir/common/imageio_avif.c.o
/home/magnus/dev/builds/c-darktable/x86_64-linux-gnu-default/src/src/common/imageio_avif.c: In function ‘dt_imageio_open_avif’:
/home/magnus/dev/builds/c-darktable/x86_64-linux-gnu-default/src/src/common/imageio_avif.c:74:12: error: too few arguments to function ‘avifImageYUVToRGB’
   74 |   result = avifImageYUVToRGB(avif, &rgb);
      |            ^~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
make[2]: *** [bin/CMakeFiles/lib_darktable.dir/build.make:1960: bin/CMakeFiles/lib_darktable.dir/common/imageio_avif.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:2431: bin/CMakeFiles/lib_darktable.dir/all] Error 2
make: *** [Makefile:171: all] Error 2
`

**Comment by TurboGit** [comment]
Happy to see SCBI used here. Can you put the full log output of the `scbi` session?

I'd like to see if you are recompiling a version if libavif or using the version packaged on the OS. TIA.

**Comment by KarlMagnusLarsson** [comment]
@TurboGit, I can send the log file. Which one and where is it? If I check in `dev/builds/.logs`, then there are many files. The file x86_64-linux-gnu-c-libavif-report.log is attached.
[x86_64-linux-gnu-c-libavif-report.log](https://github.com/darktable-org/darktable/files/9485294/x86_64-linux-gnu-c-libavif-report.log)

If I rerun `scbi --prefix=/home/magnus/test/darktable  --deps --update c-darktable` I get:

`...

2022/09/04 20:34:21 : Building c-libavif [default] (master)
2022/09/04 20:34:21 : native x86_64-linux-gnu
2022/09/04 20:34:21 : steps : setup config build install wrapup
2022/09/04 20:34:21 : get sources from git
2022/09/04 20:34:22 : no build needed, versions match
2022/09/04 20:34:22 : copy install into /home/magnus/test/darktable
2022/09/04 20:34:22 : End Building c-libavif [default] (master)
...

`

**Comment by TurboGit** [comment]
Ok, so definitely you are building libavif from sources using the master branch.

On my Debian box I'm using the native (OS package) : 

```
2022/09/04 19:57:07 : Building c-libavif [native] (master)
2022/09/04 19:57:07 : native x86_64-linux-gnu
2022/09/04 19:57:07 : steps : setup config build install wrapup
2022/09/04 19:57:07 : no build needed, versions match
2022/09/04 19:57:07 : skip global install
2022/09/04 19:57:07 : End Building c-libavif [native] (master)
```

You may try to install libavif-dev : 

`$ sudo apt install libavif-dev`

And see if it works. It should and this would mean that the current master version has some incompatibilities.

**Comment by TurboGit** [comment]
Alternatively you can just force building v0.10.1 instead of master:

`$ scbi --prefix=/home/magnus/test/darktable --deps --update c-libavif:v0.10.1 c-darktable`

**Comment by TurboGit** [comment]
@KarlMagnusLarsson : Make sure you update scbi as I have just fixed the minimum libavif version required (should not try to build master anymore).

**Comment by KarlMagnusLarsson** [comment]
Works for me!

Thank you @TurboGit, Darktable using SCBI builds and I can start darktable. [my-build-log.txt](https://github.com/darktable-org/darktable/files/9491859/my-build-log.txt)

When I execute:
`scbi --run:"darktable" c-darktable`
I get:
```2022/09/05 21:55:50 : module libexiv2-dev version v0.27.5 not found
2022/09/05 21:55:50 : c-darktable=c-exiv2 using standard variant
2022/09/05 21:55:50 : dependency libavif-dev not installed
2022/09/05 21:55:50 : c-darktable=c-libavif using standard variant
2022/09/05 21:55:50 : dependency libavif-dev not installed
2022/09/05 21:55:50 : c-libavif using standard variant
[imageio_load_module_format] failed to open `avif': (null)
```
I thought, for example that exiv2 v0.27.5 was built, but that is maybe another issue.

But, again, darktable starts. I can see CR3 metadata as well. I can also start manually, in my sandbox, `./darktable` after setting `LD_LIBRARY_PATH`.

**Comment by TurboGit** [comment]
Thanks, so indeed the issue was that you were using a too recent version of libavif. If the issue is resolved, please close this ticket. TIA.

**Comment by TurboGit** [comment]
> I thought, for example that exiv2 v0.27.5 was built, but that is maybe another issue.

It should be built yes. I don't have this message on my side. Is it displayed only when you try to load an AVIF file?

**Comment by KarlMagnusLarsson** [comment]
> It should be built yes. I don't have this message on my side. Is it displayed only when you try to load an AVIF file?
It is displayed when starting darktable. All my files used for testing are CR2 or CR3 raw from Canon. I have no AVIF files.

**Comment by KarlMagnusLarsson** [comment]
I am fine now, but closing this issue is perhaps up to originator of issue @TopRockPhoto as I am uncertain if the root cause stated in report is he same as for me.

**Comment by TopRockPhoto** [comment]
I was building AVIF from master. I will try again from tag 0.10.1 tonight.
Currently away from desktop.

Thanks for all the feedback. (Been away since I first posted the issue.
Sorry about that).

**Comment by TopRockPhoto** [comment]
Yep! I made a boo-boo. I built from main instead of from 0.10.1.

I did a proper compile of AVIF 0.10.1, then Darktable compiled. Darktable no longer crashes, Gimp no longer crashes.

I apologise for the unnecessary report. I always compile all this when doing darktable and somehow manage to mess it up this time. Will be more careful in the future.

There will be a change for avifImageYUVToRGB in avif version 1.0.0, and that change is reflected in the master branch already. So correct diagnosis, incorrect version. (Version 1.0.0 is not even in beta yet).
